src/zls.ts

```diff
@@ -0,0 +1,365 @@
+        await config.update("path", zlsBinPath, true);
+
+        return zlsBinPath;
+    });
+}
+
+export async function checkUpdateMaybe(context: ExtensionContext) {
+    const configuration = workspace.getConfiguration("zig.zls");
+    const checkForUpdate = configuration.get<boolean>("check_for_update", true);
+    if (checkForUpdate) {
+        try {
+            await checkUpdate(context, true);
+        } catch (err: any) {
+            outputChannel.appendLine(`Failed to check for update. Reason: ${err.message}`);
+        }
+    };
+}
+
+export async function startClient(context: ExtensionContext) {
+    const configuration = workspace.getConfiguration("zig.zls");
+    const debugLog = configuration.get<boolean>("debugLog", false);
+
+    const zlsPath = await getZLSPath(context);
+
+    if (!zlsPath) {
+        window.showWarningMessage("Couldn't find Zig Language Server (ZLS) executable");
+        return null;
+    }
+
+    let serverOptions: ServerOptions = {
+        command: zlsPath,
+        args: debugLog ? ["--enable-debug-log"] : []
+    };
+
+    // Options to control the language client
+    let clientOptions: LanguageClientOptions = {
+        documentSelector: [{ scheme: "file", language: "zig" }],
+        outputChannel,
+    };
+
+    // Create the language client and start the client.
+    client = new LanguageClient(
+        "zls",
+        "Zig Language Server",
+        serverOptions,
+        clientOptions
+    );
+
+    return client.start().catch(reason => {
+        window.showWarningMessage(`Failed to run Zig Language Server (ZLS): ${reason}`);
+        client = null;
+    });
+}
+
+export async function stopClient(): Promise<void> {
+    if (client) client.stop();
+    client = null;
+}
+
+// returns the file system path to the zls executable
+export async function getZLSPath(context: ExtensionContext): Promise<string | null> {
+    const configuration = workspace.getConfiguration("zig.zls");
+    var zlsPath = configuration.get<string | null>("path", null);
+
+    if (!zlsPath) {
+        zlsPath = which.sync("zls", { nothrow: true });
+    } else if (zlsPath.startsWith("~")) {
+        zlsPath = path.join(os.homedir(), zlsPath.substring(1));
+    } else if (!path.isAbsolute(zlsPath)) {
+        zlsPath = which.sync(zlsPath, { nothrow: true });
+    }
+
+    var message: string | null = null;
+
+    const zlsPathExists = zlsPath !== null && fs.existsSync(zlsPath);
+    if (zlsPath && zlsPathExists) {
+        try {
+            fs.accessSync(zlsPath, fs.constants.R_OK | fs.constants.X_OK);
+        } catch {
+            message = `\`zls.path\` ${zlsPath} is not an executable`;
+        }
+        const stat = fs.statSync(zlsPath);
+        if (!stat.isFile()) {
+            message = `\`zls.path\` ${zlsPath} is not a file`;
+        }
+    }
+
+    if (message === null) {
+        if (!zlsPath) {
+            message = "Couldn't find Zig Language Server (ZLS) executable";
```

Comment:
This is the message users will first see when installing the extension so it should be reworded to indicate that it is optional but recommended. Also it should use `showInformationMessage` since unlike the other possible messages there hasn't been any issues.